index.bs

```diff
@@ -402,9 +582,36 @@ perform the following steps:
 
 </div>
 
-Every [=bounce tracking timer period=] the user agent should run the
-[=bounce tracking timer=] algorithm given the [=wall clock=]'s
-[=wall clock/unsafe current time=].
+This algorithm is called when detecting the end of an [=extended navigation=]. This could happen
+if a user-initiated navigation is detected in [=process navigation start for bounce tracking=],
+or if the client bounce detection timer expires after [=process response received for bounce tracking=]
+without observing a client redirect.
+
+<div algorithm>
+
+To run the <dfn>record stateful bounces for bounce tracking</dfn> algorithm
+given a [=global object=] |global|, perform the following steps:
+
+1. Let |navigable| be |global|'s [=associated document=]'s [=node navigable=].
+1. [=Assert=]: |navigable| is not null.
+1. Let |topDocument| be |navigable|'s [=navigable/active document=].
+1. [=Assert=]: |topDocument| is the same as |global|'s [=associated document=].
+    This ensures that the global has not been detached from the navigable.
```

Comment:
"This ensures" isn't really right for an assertion. I think this one is guaranteed to hold, by the "the navigable whose active document is node's node document" bit of [=node navigable=], so you can just leave the Assert by itself.